src/react/hooks/use-room.ts

```diff
@@ -0,0 +1,101 @@
+import { ConnectionStatusChange, ErrorInfo, Room, RoomLifecycle, RoomStatusChange } from '@ably/chat';
+import * as Ably from 'ably';
+import { useContext, useEffect, useState } from 'react';
+
+import { ChatStatusResponse } from '../chat-status-response.js';
+import { RoomContext } from '../contexts/room-context.js';
+import { useChatConnection } from './use-chat-connection.js';
+
+/**
+ * Parameters for the useRoom hook.
+ */
+export interface UseRoomParams {
+  /**
+   * Callback for when the room status changes. The listener is removed when
+   * the component unmounts.
+   * @param change The change object.
+   */
+  onStatusChange?: (change: RoomStatusChange) => void;
+
+  /**
+   * Callback for when the connection status changes. The listener is removed
+   * when the component unmounts.
+   * @param change The change object.
+   */
+  onConnectionStatusChange?: (change: ConnectionStatusChange) => void;
+}
+
+/**
+ * The return type for the useRoom hook.
+ */
+export interface UseRoomResponse extends ChatStatusResponse {
+  /** The room object. */
+  room: Room;
+
+  /**
+   * Shortcut to room.attach. Not needed if room provider has attach=true set,
+   * which is the default behavior.
+   */
+  attach: () => Promise<void>;
+
+  /**
+   * Shortcut to room.detach. Not needed if room provider has attach=true
+   * set, which is the default behavior.
+   */
+  detach: () => Promise<void>;
+}
+
+/**
+ * Hook that provides access to the current room.
+ *
+ * @param params Register optional callbacks, see {@link UseRoomParams}.
+ * @returns {@link UseRoomResponse}
+ */
+export const useRoom = (params?: UseRoomParams): UseRoomResponse => {
+  const context = useContext(RoomContext);
+
+  if (!context) {
+    throw new Ably.ErrorInfo('useRoom hook must be used within a chat RoomProvider', 40000, 400);
```

Comment:
```suggestion
    throw new Ably.ErrorInfo('useRoom hook must be used within a ChatRoomProvider', 40000, 400);
```

Reply:
Do we want `ChatRoomProvider` or `RoomProvider`? DR is `ChatRoomProvider`. I don't have a strong view either way.

I thought we have `ChatClientProvider` because we have the object called `ChatClient`, so I went with `RoomProvider` because the object is `Room`.

Reply:
The reason I went for `ChatRoomProvider` in the DR is because I could imagine that someone might want to use `RoomProvider` in their own app, so this disambiguated

Reply:
If we rename this, I assume we must rename the provider file and all imports?

Reply:
Yes